Make each Bundle Analyzer report open in new tab with the proper title

kmanijak opened the pull request:
There are currently five building processes. When running `npm run analyze-bundles`, five reports are generated. However one report overrides the previous one (in the same tab/port) and it's time dependant. Whichever finishes the last, this one will be displayed as final.

Looking at the reports through this process is confusing as it seems it evolves, rather then overrides and the final report may different between runs (in fact those are simply different reports you may see, but it may look like different build result).

This PR makes each Bundle Analyzer report open in new tab on a separate port and with a proper title: Core, Main, Frontend, Payments, Extensions (see videos below).

### Screenshots

| Before | After |
| ------ | ----- |
|   https://user-images.githubusercontent.com/20098064/209946491-0b316966-7291-431e-8ee4-56c6dba45e91.mov |   https://user-images.githubusercontent.com/20098064/209946425-cc9dc092-3c15-4c73-995b-8265dff0adb5.mov  |






#### User Facing Testing

1. When in project, run `npm run analyze-bundles`

**Expected:** Bundle Analyzer reports are open in 5 separate tabs, each named accordingly: Core, Main, Frontend, Payments, Extensions

* [x] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested by users (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

kmanijak committed:
Make each bundle analzyer report open in new tab with the proper title

dinhtungdu commented on bin/webpack-configs.js:
```diff
@@ -584,7 +590,7 @@ const getExtensionsConfig = ( options = {} ) => {
 		},
 		plugins: [
-			...sharedPlugins,
+			...getSharedPlugins( { bundleAnalyzerReportTitle: 'Extensions' } ),
```
I think we should use Experimental Extensions here. We have Payment Method Extensions above, so we should be more explicit.

kmanijak replied:
Thanks @dinhtungdu , you're right, I totally agree. I'll adjust the report titles:
- Extensions -> Experimental Extensions
- Payments -> Payment Method Extensions
so it aligns with Progress Plugin and are distinguishable!

dinhtungdu reviewed (approved):
This is pretty neat @kmanijak. I left a minor comment but pre-approving. 🚀

kmanijak committed:
Change Bundle Analyzer report titles to better reflect what they represent

Change from Extensions to Experimental Extensions and from Payments to Payment Method Extensions

kmanijak committed:
Merge branch 'trunk' into add/allow-to-open-multiple-bundle-analyzer-reports

kmanijak commented:
Merging with the bypass - E2E tests with Gutenberg fail, the same as on trunk. All the other checks are passing, including E2E without Gutenberg.
